controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -464,18 +458,36 @@ func (r *StorageConsumerReconciler) reconcileCephRadosNamespace(
 		}
 		rns.Namespace = r.namespace
 
-		if _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rns, func() error {
-			if err := controllerutil.SetControllerReference(r.storageConsumer, rns, r.Scheme); err != nil {
-				return err
+		if bp.DeletionTimestamp.IsZero() {
+			if _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rns, func() error {
+				if err := controllerutil.SetControllerReference(r.storageConsumer, rns, r.Scheme); err != nil {
+					return err
+				}
+				if err := controllerutil.SetOwnerReference(additionalOwner, rns, r.Scheme); err != nil {
+					return err
+				}
+				rns.Spec.Name = radosNamespaceName
+				rns.Spec.BlockPoolName = bp.Name
+				return nil
+			}); err != nil {
+				multierr.AppendInto(&combinedErr, err)
 			}
-			if err := controllerutil.SetOwnerReference(additionalOwner, rns, r.Scheme); err != nil {
-				return err
+		} else {
+			if bp.Status != nil {
+				for idx := range bp.Status.Conditions {
```

Comment:
I think it will make the code clearer if we are to search for the condition using `slices.Index` or `slices.IndexFunc`

Reply:
done.